server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -45,31 +42,9 @@
 import static java.time.temporal.ChronoField.MONTH_OF_YEAR;
 import static java.time.temporal.ChronoField.NANO_OF_SECOND;
 import static java.time.temporal.ChronoField.SECOND_OF_MINUTE;
-import static org.elasticsearch.common.util.ArrayUtils.prepend;
 
 public class DateFormatters {
 
-    /**
-     * The ISO8601 parser is as close as possible to the java.time based parsers, but there are some strings
-     * that are no longer accepted (multiple fractional seconds, or multiple timezones) by the ISO parser.
-     * If a string cannot be parsed by the ISO parser, it then tries the java.time one.
-     * If there's lots of these strings, trying the ISO parser, then the java.time parser, might cause a performance drop.
-     * So provide a JVM option so that users can just use the java.time parsers, if they really need to.
-     * <p>
-     * Note that this property is sometimes set by {@code ESTestCase.setTestSysProps} to flip between implementations in tests,
-     * to ensure both are fully tested
-     */
-    @UpdateForV9(owner = UpdateForV9.Owner.CORE_INFRA) // evaluate if we need to deprecate/remove this
-    private static final boolean JAVA_TIME_PARSERS_ONLY = Booleans.parseBoolean(System.getProperty("es.datetime.java_time_parsers"), false);
```

Comment:
One Q: by removing this setting, will cluster that have explicitly set this (to either true or false) just ignore it or fail to start? If it's the latter, should we deprecate the setting instead, or add `>breaking` to the PR?

Reply:
It's a JVM option, so it'll just be ignored